### PR TITLE
fix(core): Handle scoop aliases and broken(edited,copied) shim

### PR DESCRIPTION
**Pull request by HUMORCE**
#### Motivation and Context

`scoop-which` cannot get path when these case:
- `scoop which <manually creaetd/edited shim missing shimTarget>`
- `scoop which <copied shim>` e.g. moving Scoop root directory from other locations
- `scoop which scoop-<scoop_alias>`

#### How Has This Been Tested?

Screenshot_230620033720

#### Checklist:

- [x] I have read the [Contributing Guide](https://github.com/ScoopInstaller/.github/blob/main/.github/CONTRIBUTING.md).
- [x] I have ensured that I am targeting the `develop` branch.
- [ ] I have updated the documentation accordingly.
- [ ] I have updated the tests accordingly.
- [x] I have added an entry in the CHANGELOG.

### CODE REVIEW

**Comment by rashil2000** [comment]
Looks good, update changelog?

**Comment by HUMORCE** [comment]
Done.

File: CHANGELOG.md [hunk 1]
```diff
@@ -23,6 +23,7 @@
 - **scoop-info:** Fix errors in file size collection when `--verbose` ([#5352](https://github.com/ScoopInstaller/Scoop/pull/5352))
 - **shim:** Use bash executable directly ([#5433](https://github.com/ScoopInstaller/Scoop/issues/5433))
 - **scoop-checkup:** Skip defender check in Windows Sandbox ([#5519]https://github.com/ScoopInstaller/Scoop/issues/5519)
+- **core:** Handle scoop aliases and broken(edited,copied) shim ([#5551]https://github.com/ScoopInstaller/Scoop/issues/5551)
```
Comment (rashil2000) [inline CHANGELOG.md:26]:
```suggestion
- **core:** Handle scoop aliases and broken(edited,copied) shim ([#5551](https://github.com/ScoopInstaller/Scoop/issues/5551))
```

**Comment by rashil2000** [comment]
Also there are conflicts.

**Pull request merged**